llama_index/response_synthesizers/google/generativeai/base.py

```diff
@@ -0,0 +1,265 @@
+
+    def synthesize(
+        self,
+        query: QueryTextType,
+        nodes: List[NodeWithScore],
+        additional_source_nodes: Optional[Sequence[NodeWithScore]] = None,
+        **response_kwargs: Any,
+    ) -> Response:
+        """Returns a grounded response based on provided passages.
+
+        Returns:
+            Response's `source_nodes` will begin with a list of attributed
+            passages. These passages are the ones that were used to construct
+            the grounded response. These passages will always have no score,
+            the only way to mark them as attributed passages. Then, the list
+            will follow with the originally provided passages, which will have
+            a score from the retrieval.
+
+            Response's `metadata` may also have have an entry with key
+            `answerable_probability`, which is the model's estimate of the
+            probability that its answer is correct and grounded in the input
+            passages.
+        """
+        if len(nodes) == 0:
+            return Response("Empty Response")
+
+        if isinstance(query, str):
+            query = QueryBundle(query_str=query)
+
+        with self._callback_manager.event(
+            CBEventType.SYNTHESIZE, payload={EventPayload.QUERY_STR: query.query_str}
+        ) as event:
+            internal_response = self.get_response(
+                query_str=query.query_str,
+                text_chunks=[
+                    n.node.get_content(metadata_mode=MetadataMode.LLM) for n in nodes
+                ],
+                **response_kwargs,
+            )
+
+            additional_source_nodes = list(additional_source_nodes or [])
+
+            external_response = self._prepare_external_response(
+                internal_response, nodes + additional_source_nodes
+            )
+
+            event.on_end(payload={EventPayload.RESPONSE: external_response})
+
+        return external_response
+
+    async def asynthesize(
+        self,
+        query: QueryTextType,
+        nodes: List[NodeWithScore],
+        additional_source_nodes: Optional[Sequence[NodeWithScore]] = None,
+        **response_kwargs: Any,
+    ) -> Response:
+        # TODO: Implement a true async version.
+        return self.synthesize(query, nodes, additional_source_nodes, **response_kwargs)
+
+    def _prepare_external_response(
+        self,
+        response: SynthesizedResponse,
+        source_nodes: List[NodeWithScore],
+    ) -> Response:
+        return Response(
+            response=response.answer,
+            source_nodes=[
+                NodeWithScore(node=TextNode(text=passage))
+                for passage in response.attributed_passages
+            ]
+            + source_nodes,
+            metadata={
+                "answerable_probability": response.answerable_probability,
+            },
+        )
+
+    def _get_prompts(self) -> PromptDictType:
```

Comment:
could remove those `_get_prompts` and `_update_prompts `

Reply:
If I remove them, I get this error message: Can't instantiate abstract class GoogleTextSynthesizer with abstract methods _get_prompts, _update_prompts

Looks like an implementation is needed for those methods.